salt/haproxy/init.sls

```diff
@@ -0,0 +1,48 @@
+---
+/etc/haproxy/haproxy.cfg:
+  file.managed:
+    - source: salt://haproxy/haproxy.cfg.jinja
+    - template: jinja
+    - user: root
+    - group: root
+    - mode: 644
+    - makedirs: True
+    - dir_mode: 755
+
+haproxy:
+  file.managed:
+    - name: /etc/kubernetes/manifests/haproxy.yaml
+    - source: salt://haproxy/haproxy.manifest
+    - template: jinja
+    - user: root
+    - group: root
+    - mode: 644
+    - makedirs: True
+    - dir_mode: 755
+{% if "kube-master" in salt['grains.get']('roles', []) %}
+  iptables.append:
+{% else %}
+  iptables.delete:
```

Comment:
While it's probably harmless to do so, we're not listening on 6443 on the workers, which would leave an open and unbound port - which I don't like the idea of :)

We could just listen on 6443 everywhere, but I think it's easier to ask people to aim the public FQDNs that compose the core service APIs to the reasonably fixed pool or masters, rather than the hopefully fluid pool of workers.

I don't have a strong opinion either way - other than -  we should only be exposing ports where we actually have a service listening on that machine